### PR TITLE
346 runs history queries 12 runs history collapsed view

### LINKED ISSUE #346: Runs history queries (1/2) - runs history collapsed view
### User story
As a user monitoring my workflows, I would like to determine the final status of a WorkOrder run, so that I know if every step has completed successfully. 

### Details
Given a project ID and a WorkOrder, return the final status of a workflow run. 

### Implementation notes
See this document for notes: https://docs.google.com/spreadsheets/d/1YvvskT8pJckTVbc18JW0qo38pNAjlgLp2OM7ZRx2QCk/edit#gid=0

### Tests
Given a workflow with Job A, B and C. 
When run A,B and C are successful, I expect to see a wokflow run status of 'Success'. 
When run A and B are successful but C fails, I expect to see a wokflow run status of 'Failure'. 
When run A times out, I expect to see a wokflow run status of 'Timeout'. 
When run B crashed, I expect to see a wokflow run status of 'Crashed'. 

@taylordowns2000 should this implementation be returning status codes that then get changed in the UI or the actual words 'Success', 'Failure' etc ?

### PR DESCRIPTION
**Pull request by wanecode**
## Related issue

#347 Run history queries 2/2

Fixes #346 Runs history queries 1/2

## Checklist before requesting a review

- [x] I have performed a self-review of my code
- [ ] If needed, I've updated the changelog
- [x] Amber has QA'd this feature

### CODE REVIEW

File: lib/lightning_web/live/run_live/components.ex [hunk 1]
```diff
@@ -100,21 +30,24 @@ defmodule LightningWeb.RunLive.Components do
...
         class="my-auto whitespace-nowrap p-6 font-medium text-gray-900 dark:text-white"
       >
-        workFlowName
+        <%= @workflow.name %>
       </th>
+      <td class="my-auto p-6"><%= @reason.dataclip_id %></td>
       <td class="my-auto p-6">12i78iy</td>
-      <td class="my-auto p-6">UTC 24:20:60</td>
+      <td class="my-auto p-6">
+        <%= @last_attempt.last_run.finished_at |> Calendar.strftime("%c") %>
```
Comment (stuartc) [inline lib/lightning_web/live/run_live/components.ex:43]: I managed to get the app to throw an exception with runs that haven't finished. I've fixed the issue in the WorkOrder component, but not here.

We need a test for this.

Also the name of this function component is `work_order`, should it not be attempt or something else? I'm confused by the `Components.WorkOrder` and this?

Is this file being used anymore? What about `AttemptComponent`? Please remove this file is we're not using it.

**Review by stuartc — CHANGES_REQUESTED** [review]
This is great overall @wanecode, really great. Could you address some of my comments.

I have done a tiny bit of changes here: 9802577541dab6cc681e5fcf1316a0a55fbcdd63

- Added `pending` status for Runs that are in progress
- Removed use of Pipeline to make fixtures for WorkOrderLive tests
- Remove codecov skip comments for RunLive WorkOrder components

File: lib/lightning_web/live/run_live/components.ex (hunk 1 above)
Comment (wanecode) [reply lib/lightning_web/live/run_live/components.ex:43]: Only pill function are needed in this file. Removed work_order function

File: lib/lightning_web/live/run_live/attempt_component.ex [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+              <%= case @last_run.exit_code do %>
+                <% val when val > 0-> %>
+                  <span class="my-auto ml-2 whitespace-nowrap rounded-full bg-red-200 py-2 px-4 text-center align-baseline text-xs font-medium leading-none text-red-800">
+                    Failure
+                  </span>
+                <% val when val == 0 -> %>
+                  <span class="my-auto ml-2 whitespace-nowrap rounded-full bg-green-200 py-2 px-4 text-center align-baseline text-xs font-medium leading-none text-green-800">
+                    Success
+                  </span>
+                <% _ -> %>
```
[the remark this answers is not in the document]
Comment (wanecode) [reply lib/lightning_web/live/run_live/attempt_component.ex:46]: I think we should have it here too to keep things in sync

**Pull request merged**